lib/sockets/pub.js

```diff
@@ -38,13 +38,10 @@ PubSocket.prototype.__proto__ = Socket.prototype;
 PubSocket.prototype.send = function(msg){
   var socks = this.socks;
   var len = socks.length;
-  var sock;
-
   var buf = this.pack(arguments);
 
-  for (var i = 0; i < len; i++) {
-    sock = socks[i];
-    if (sock.writable) sock.write(buf);
+  for (var sock of socks) {
+      if (sock.writable) sock.write(buf);
```

Comment:
should set to if(sock && sock.writable)... because sometimes sock comes null/undefined.